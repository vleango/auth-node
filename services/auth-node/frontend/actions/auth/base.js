export const ROOT_API_URL = `${process.env.ROOT_URI}/api`

export const access = (token) => ({
  type: 'ACCESS',
  token
});

export const logout = () => ({
  type: 'LOGOUT'
});
