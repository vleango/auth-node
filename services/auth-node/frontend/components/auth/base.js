import React from 'react';
import Validator from 'validator';
import { pick } from 'lodash';
import FacebookLogin from 'react-facebook-login';

export const PROVIDER_ERR_MSG = 'Error using provider';

export class AuthComponentBase extends React.Component {

  FACEBOOK_APP_ID = `${process.env.FACEBOOK_APP_ID}`;

  constructor(props) {
    super(props);
  }

  onInputChange = (e) => {
    const field = e.target.name;
    const value = e.target.value;
    this.setState(() => ({ [field]: value }));
  }

  isValidEmailPasswordLength() {
    const {email, password} = this.state;
    return Validator.isEmail(email) && password.length >= 8;
  }

  isValidPasswordMatch() {
    const {password, password_confirmation} = this.state;
    return password === password_confirmation;
  }

  authorizeWithFacebook = async (response) => {
    const values = pick(response, ['name', 'email', 'accessToken', 'userID']);
    if(values.accessToken && values.name && values.email && values.userID) {
      const [first_name, last_name] = response.name.split(' ');
      try { await this.props.startAccess('provider', { ...values, first_name, last_name, access: 'facebook' }) }
      catch(error) { await this.setState(() => ({ error: PROVIDER_ERR_MSG, submitting: false })) }
    }
    else {
      console.log('Facebook Login Cancelled');
    }
  }

  render() { }

}
