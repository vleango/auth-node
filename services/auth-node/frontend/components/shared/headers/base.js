import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

export const Header = ({ headerLink, onClickPressed, buttonText }) => (
  <div>
    <header>
      <Link to={headerLink}>
        <h3>App</h3>
      </Link>
      <button onClick={onClickPressed}>{buttonText}</button>
    </header>
  </div>
);

export default Header;
