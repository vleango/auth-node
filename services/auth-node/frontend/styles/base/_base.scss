html {
  font-size: 62.5%; // 16px * .625 = 10 (base 10)
}

body {
  font-family: Helvetica, Arial, sans-serif;
  font-size: $m-size;
}

button {
  cursor: pointer;
}

button:disabled {
  cursor: default;
}
