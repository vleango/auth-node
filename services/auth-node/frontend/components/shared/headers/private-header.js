import React from 'react';
import { connect } from 'react-redux';

import Header from './base';
import { startLogout } from '../../../actions/auth/logout';

export const PrivateHeader = ({ startLogout }) => (
  <Header
    headerLink='/dashboard'
    buttonText='Logout'
    onClickPressed={startLogout} />
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: async () => await dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(PrivateHeader);
