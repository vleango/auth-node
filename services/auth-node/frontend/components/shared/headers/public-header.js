import React from 'react';

import Header from './base';

export class PublicHeader extends React.Component {

  loginBtnClicked = () => {
    this.props.history.push('/login');
  };

  render() {
    return (
      <Header
        headerLink='/'
        buttonText='Login'
        onClickPressed={this.loginBtnClicked} />
    );
  }

}

export default PublicHeader;
