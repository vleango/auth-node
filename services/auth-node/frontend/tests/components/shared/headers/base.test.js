import React from 'react';
import { shallow } from 'enzyme';

import { Header } from '../../../../components/shared/headers/base';

describe('Components', () => {
  describe('Shared', () => {
    describe('Headers', () => {
      describe('Base', () => {

        it('should render Header correctly', () => {
        	const wrapper = shallow(<Header headerLink='/' />);
        	expect(wrapper).toMatchSnapshot();
        });

        it('should call action on button click', () => {
        	const action = jest.fn();
        	const wrapper = shallow(<Header headerLink='/' onClickPressed={action} />);
        	wrapper.find('button').simulate('click');
        	expect(action).toHaveBeenCalled();
        });

      });

    });
  });
});
