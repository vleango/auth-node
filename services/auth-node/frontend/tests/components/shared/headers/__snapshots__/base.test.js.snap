// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Shared Headers Base should render Header correctly 1`] = `
<div>
  <header>
    <Link
      replace={false}
      to="/"
    >
      <h3>
        App
      </h3>
    </Link>
    <button />
  </header>
</div>
`;
